buildSrc/src/main/groovy/org/gradle/build/UpdateReleasedVersions.groovy

```diff
@@ -21,24 +21,43 @@ import com.google.common.base.Preconditions
 import groovy.json.JsonBuilder
 import groovy.json.JsonSlurper
 import org.gradle.api.DefaultTask
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.util.GradleVersion
 
+import javax.inject.Inject
+
 class UpdateReleasedVersions extends DefaultTask {
 
     @Internal
     File releasedVersionsFile
 
     @Internal
-    ReleasedVersion currentReleasedVersion
+    Property<ReleasedVersion> currentReleasedVersion
+
+    @Inject
+    UpdateReleasedVersions(ObjectFactory objectFactory) {
+        currentReleasedVersion = objectFactory.property(ReleasedVersion)
+    }
+
+    void setReleasedVersion(Provider<ReleasedVersion> provider) {
+        currentReleasedVersion.set(provider)
+    }
+
+    void setReleasedVersion(ReleasedVersion releasedVersion) {
+        currentReleasedVersion.set(releasedVersion)
+    }
 
     @TaskAction
     void updateVersions() {
+        ReleasedVersion currentReleasedVersionValue = currentReleasedVersion.get()
         Preconditions.checkNotNull(releasedVersionsFile, "File to update not specified")
-        Preconditions.checkNotNull(currentReleasedVersion, "No currentReleasedVersion property set")
+        Preconditions.checkNotNull(currentReleasedVersionValue, "No currentReleasedVersion property set")
```

Comment:
I guess this cannot be null since a provider always returns non-null values, right? `currentReleasedVersion.get()` would throw an exception if it is not set.

Reply:
That's true. Thought about it for a second while implementing and then let it go. ;-) I guess I'll need to write a test for invoking the task action which doesn't exist yet.

Reply:
Yeah, `updateReleasedVersions` is static, so I can test it and since it doesn't use any fields.

Regarding `@Input`: The task is modifying the `releasedVersionsFile`. This file would be an input and an output. It is not possible to do that currently. This is why I opted to not annotating anything. Do you have any ideas how to solve this problem?